fvm/environment/block_info.go

```diff
@@ -142,6 +142,10 @@ func (info *blockInfo) GetBlockAtHeight(
 		return runtimeBlockFromHeader(info.blockHeader), true, nil
 	}
 
+	if height+uint64(flow.DefaultTransactionExpiry) < info.blockHeader.Height {
```

Comment:
When creating protocol snapshot, we are already adding `X` number of extra blocks below the root block to the protocol snapshot, where `X` is [flow.DefaultTransactionExpiry](https://github.com/onflow/flow-go/blob/master/model/flow/sealing_segment.go#L36C11-L36C22), which currently is set to [600](https://github.com/onflow/flow-go/blob/master/model/flow/constants.go#L22). 

So all we need is to limit the API to not go beyond that limit, as long as the queried height is within `DefaultTransactionExpiry` number of blocks, then the block must have been indexed already, and available to be queried.

Reply:
The actual transaction expiry would only be [less than the default transaction expiry. ](https://github.com/onflow/flow-go/blob/leo/6939-limit-get-block-height/module/builder/collection/build_ctx.go#L51)

So it's safe and simple to use the default.